### ISSUE TITLE
Help getting started

### ISSUE THREAD

**Issue by jsejcksn** [AtomLinter/linter-eslint#736]
I read the readme at [the Atom package page](https://atom.io/packages/linter-eslint) and I'm still not sure why it's not working for me.

I installed the package and [linter](https://atom.io/packages/linter) and added this line in the settings:

> **.eslintrc Path**
> It will only be used when there's no config file in the project
> `/Users/jesse/Applications/npm/.eslintrc.json`

My referenced `.eslintrc.json` file looks like this:

``` json
{
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "node": true
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ]
  }
}
```

I restarted Atom and now everything should be working, right? When I open a javascript file and add a this line:

``` js
test() bad.code = "not working"
```

I don't get any warnings or errors...

**Comment by IanVS** [comment]
Hi, can you try running the `Linter Eslint: debug` command, and paste the output here?

**Comment by jsejcksn** [comment]
Hi @IanVS—thanks for responding. Here's that output:

```
atom version: 1.11.1
linter-eslint version: 8.0.0
eslint version: 3.7.1
hours since last atom restart: 0.4
platform: darwin
Using bundled fallback eslint from /Users/jesse/.atom/packages/linter-eslint/node_modules/eslint
linter-eslint configuration: {
  "eslintrcPath": "/Users/jesse/Applications/npm/.eslintrc.json",
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": []
}
```

**Comment by IanVS** [comment]
The line `Using bundled fallback eslint` means you don't have eslint installed anywhere on your system, and this package is falling back to its bundled version.  That's usually not what you want, and we would recommend you install `eslint` into your project, or less ideally, install it globally.  

Furthermore, unless you have a good reason to put your `.eslintrc.json` file where it is, I would recommend either putting it directly in your home directory (in which case it will apply to every project), or even better into the root directory of the project you're working on (since each project can have a different configuration).

To answer your original question, I have a feeling that `disableWhenNoEslintConfig` should be turned off, as that is likely what is disabling linting for your file.

**Comment by jsejcksn** [comment]
@IanVS 

**Re: eslint install**: I often work in individual javascript files not related to a project and intend to use this package for that purpose. I would prefer not to manage a global installation of eslint for that reason, and I should be able to rely on the package providing a (reasonably) updated version of eslint, right?

**Re: rc file**: Isn't that what `eslintrcPath` is for? To specify where the package should look for my `.eslintrc.json` file? I am trying to understand how it works, so can you please confirm? Is there something wrong with my setup that is causing the package to not find my  `.eslintrc.json` file?

Your suspicion was correct about `disableWhenNoEslintConfig` causing the problem. Isn't it ok to have that setting if my `eslintrcPath` is correct?

**Comment by IanVS** [comment]
For your use case you can indeed use the bundled eslint version (the version it installed is the latest), but I'd still recommend installing a global ESLint if you're able.  You'll just have more control over it, is all.

Yes, you can use `eslintrcPath`, if that's how you'd like to do it.  I just wanted to make you aware that you could also put it in your home, and then it will be picked up when running `eslint` from the command line as well (if you install eslint globally, that is).  

The purpose of `disableWhenNoEslintConfig` is to avoid getting spammed with lint errors when you open a file in a project that doesn't have eslint configured.  It does not consider the manually specified `eslintrcPath` to be a found configuration.  By setting `eslintrcPath`, you're basically saying that _all_ files should follow that configuration, which it sounds like you want.  Therefore, there's no reason to turn on `disableWhenNoEslintConfig`.  Does that make any sense?

**Comment by IanVS** [comment]
I'm closing the issue since it sounds like linting is working for you.  Feel free to keep discussing here though.

**Comment by jsejcksn** [comment]
That does make sense—thanks! I don't remember seeing anything about that in the settings or the readme—maybe they should mention the exclusivity of that interaction between `eslintrcPath` and `disableWhenNoEslintConfig `.

**Re: home/global install**: I actually did initially install eslint globally and ran `eslint --init` and that's how I got my `.eslintrc.json` file. That directory is my npm `prefix` so that's where eslint would look (instead of my home folder) if I install it again globally, right?

Thanks, again, for the guidance, @IanVS.
